src/metadata/trackmetadatataglib.cpp

```diff
@@ -0,0 +1,688 @@
+    }
+}
+
+void readID3v2Tag(TrackMetadata* pTrackMetadata,
+        const TagLib::ID3v2::Tag& tag) {
+
+    // Print every frame in the file.
+    if (kDebugMetadata) {
+        for (TagLib::ID3v2::FrameList::ConstIterator it(
+                tag.frameList().begin()); it != tag.frameList().end(); ++it) {
+            qDebug() << "ID3V2" << (*it)->frameID().data() << "-"
+                    << toQString((*it)->toString());
+        }
+    }
+
+    readTag(pTrackMetadata, tag);
+
+    const TagLib::ID3v2::FrameList bpmFrame(tag.frameListMap()["TBPM"]);
+    if (!bpmFrame.isEmpty()) {
+        pTrackMetadata->setBpmString(toQStringFirst(bpmFrame));
+    }
+
+    const TagLib::ID3v2::FrameList keyFrame(tag.frameListMap()["TKEY"]);
+    if (!keyFrame.isEmpty()) {
+        pTrackMetadata->setKey(toQStringFirst(keyFrame));
+    }
+
+    // Foobar2000-style ID3v2.3.0 tags
+    // TODO: Check if everything is ok.
+    TagLib::ID3v2::FrameList textFrames(tag.frameListMap()["TXXX"]);
+    for (TagLib::ID3v2::FrameList::ConstIterator it(textFrames.begin());
+            it != textFrames.end(); ++it) {
+        TagLib::ID3v2::UserTextIdentificationFrame* replaygainFrame =
+                dynamic_cast<TagLib::ID3v2::UserTextIdentificationFrame*>(*it);
+        if (replaygainFrame && replaygainFrame->fieldList().size() >= 2) {
+            const QString desc(
+                    toQString(replaygainFrame->description()).toLower());
+            if (desc == "replaygain_album_gain") {
+                pTrackMetadata->setReplayGainDbString(
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+            // Prefer track gain over album gain.
+            if (desc == "replaygain_track_gain") {
+                pTrackMetadata->setReplayGainDbString(
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+        }
+    }
+
+    const TagLib::ID3v2::FrameList albumArtistFrame(tag.frameListMap()["TPE2"]);
+    if (!albumArtistFrame.isEmpty()) {
+        pTrackMetadata->setAlbumArtist(toQStringFirst(albumArtistFrame));
+    }
+
+    if (pTrackMetadata->getAlbum().isEmpty()) {
+        const TagLib::ID3v2::FrameList originalAlbumFrame(
+                tag.frameListMap()["TOAL"]);
+        pTrackMetadata->setAlbum(toQStringFirst(originalAlbumFrame));
+    }
+
+    const TagLib::ID3v2::FrameList composerFrame(tag.frameListMap()["TCOM"]);
+    if (!composerFrame.isEmpty()) {
+        pTrackMetadata->setComposer(toQStringFirst(composerFrame));
+    }
+
+    const TagLib::ID3v2::FrameList groupingFrame(tag.frameListMap()["TIT1"]);
+    if (!groupingFrame.isEmpty()) {
+        pTrackMetadata->setGrouping(toQStringFirst(groupingFrame));
+    }
+
+    // ID3v2.4.0: TDRC replaces TYER + TDAT
+    const TagLib::ID3v2::FrameList recordingDateFrame(
+            tag.frameListMap()["TDRC"]);
+    if (!recordingDateFrame.isEmpty()) {
+        pTrackMetadata->setYear(toQStringFirst(recordingDateFrame));
+    }
```

Comment:
I think we should fallback to TYER if TDRC is empty 
http://id3.org/id3v2.4.0-changes

TYER can be up to "yyyy-MM-ddTHH:mm:ss" format 
At least the release month could be significant. 
Does Mixxx already support the full format?  If yes, we should rename the setYear() chain. 
Otherwise we should be sure nothing bad happens when we find a track with more info and place a TODO